frontend/src/components/Dashboard/SideBar.js

```diff
@@ -0,0 +1,261 @@
+            minWidth: '85%',
+            maxWidth: '85%',
+        },
+    };
+};
+
+const BorderLinearProgress = withStyles({
+    root: {
+        height: 10,
+        width: 600,
+        backgroundColor: lighten('#3577cb', 0.5),
+    },
+    bar: {
+        borderRadius: 20,
+        backgroundColor: '#3577cb',
+    },
+})(LinearProgress);
+
+class SideBar extends React.Component {
+    /** Constructor. */
+    constructor(props) {
+        super(props);
+        this.state = {
+            bundles: [],
+        };
+    }
+
+    componentDidMount() {
+        const { classes } = this.props;
+        // Fetch bundles' count in different states owned by the current user one by one
+        const states: String[] = [
+            'uploading',
+            'created',
+            'staged',
+            'making',
+            'starting',
+            'preparing',
+            'running',
+            'finalizing',
+            'ready',
+            'failed',
+            'killed',
+            'worker_offline',
+        ];
+        const bundleUrl: URL = '/rest/interpret/search';
+        const fetchBundles = (stateIndex, bundlesDict) => {
+            $.ajax({
+                url: bundleUrl,
+                dataType: 'json',
+                type: 'POST',
+                cache: false,
+                data: JSON.stringify(
+                    stateIndex < states.length
+                        ? {
+                              keywords: [
+                                  'owner=' + this.props.userInfo.user_name,
+                                  '.count',
+                                  'state=' + states[stateIndex],
+                              ],
+                          }
+                        : {
+                              keywords: [
+                                  'owner=' + this.props.userInfo.user_name,
+                                  '.count',
+                                  '.floating',
+                              ],
+                          },
+                ),
+                contentType: 'application/json; charset=utf-8',
+                success: (data) => {
+                    if (stateIndex < states.length) {
+                        bundlesDict[states[stateIndex]] = data.response.result;
+                        fetchBundles(stateIndex + 1, bundlesDict);
+                    } else {
+                        bundlesDict['floating'] = data.response.result;
+                        const bundles: HTMLElement[] = [];
+                        for (let state in bundlesDict) {
+                            // Only show the non-zero items
+                            if (bundlesDict[state] > 0) {
+                                bundles.push(
+                                    <li key={state} className={classes.subheader}>
+                                        <a
+                                            className={classes.subheader}
+                                            href={
```

Comment:
I don't think we should link to the code -- that's not the best experience for the user, and the code on master can also change. Maybe we can make a help dialog / tooltip that shows up. For now, we could also just remove the link.

Reply:
I see. Done